pkg/apis/noobaa/v1alpha1/noobaa_types.go

```diff
@@ -197,6 +197,20 @@ type NooBaaSpec struct {
 	// to true
 	// +optional
 	LoadBalancerSourceSubnets LoadBalancerSourceSubnetSpec `json:"loadBalancerSourceSubnets,omitempty"`
+
+	// Configuration related to autoscaling
+	// +optional
+	Autoscaler AutoscalerSpec `json:"autoscaler,omitempty"`
+}
+
+// AutoscalerSpec defines different actoscaling spec such as autoscaler type and prometheus namespace
+type AutoscalerSpec struct {
+	// Type of autoscaling (optional) for noobaa-endpoint, hpav1 - default kubernetes based, hpav2 and keda - Prometheus metrics based
+	// +kubebuilder:validation:Enum=hpav1;hpav2;keda
+	AutoscalerType AutoscalerTypes `json:"autoscalerType"`
+
+	// Prometheus namespace that scrap metrics from noobaa
+	PrometheusNamespace string `json:"prometheusNamespace"`
```

Comment:
`PrometheusNamespace` should be optional
```suggestion
	// Prometheus namespace that scrap metrics from noobaa
	// +optional
	PrometheusNamespace string `json:"prometheusNamespace"`
```